tests/utils/test_visitor_helper.py

```diff
@@ -481,3 +483,93 @@ class MyClass:
         checkArgDefaults=False,
     )
     assert extracted == expected
+
+src1 = '''
+class SimpleInlineDoc:
+    """A class for testing and experimenting with code snippets."""
+
+    field1: int = 5
+    """Inline documentation for classvar."""
+'''
+
+src2 = '''
+class InlineDocAfter:
+    """Another class for testing."""
+
+    field1: int = 10
+    """Inline documentation for classvar."""
+
+    field2: str
+    """Inline documentation for an instance variable."""
+
+    field3: int = 5
+    """Inline documentation for another classvar."""
+'''
+
+src3 = '''
+class MixedAttributeDoc:
+    """
+    A class with mixed attribute documentation styles.
+
+    {attribute_documentation}Documentation for field2 using the attribute directive.
+    """
+
+    field1: int = 42
+    """Documentation for field1 placed after the declaration."""
+
+    field2: str = "str"
+'''
+
+
+@pytest.mark.parametrize(
+    ("src", "style", "attribute_documentation"),
+    [
+        (src1, "sphinx", None),
+        (src1, "google", None),
+        (src1, "numpy", None),
+        (src2, "sphinx", None),
+        (src2, "google", None),
+        (src2, "numpy", None),
```

Comment:
Nit: an elegant way to improve readability is to use `textwrap.dedent`, such as:

```python
from textwrap import dedent  # textwrap is python stdlib

@pytest.mark.parametrize(
    ("src", "style", "attribute_documentation"),
    [
        (
            dedent(
            '''
            class SimpleInlineDoc:
            """A class for testing and experimenting with code snippets."""

            field1: int = 5
            """Inline documentation for classvar."""
            '''
            ), "sphinx", None),
    ...
```

This way, you don't need to first define `src1`, `src2`, etc. outside.